scene/2d/gpu_particles_2d.cpp

```diff
@@ -140,6 +140,62 @@ void GPUParticles2D::set_process_material(const Ref<Material> &p_material) {
 	update_configuration_warnings();
 }
 
+void GPUParticles2D::set_enable_trail(bool p_enabled) {
+	trail_enabled = p_enabled;
+	RS::get_singleton()->particles_set_trails(particles, trail_enabled, trail_length);
+	update_configuration_warnings();
+	update();
+
+	RS::get_singleton()->particles_set_transform_align(particles, p_enabled ? RS::PARTICLES_TRANSFORM_ALIGN_Y_TO_VELOCITY : RS::PARTICLES_TRANSFORM_ALIGN_DISABLED);
+}
+void GPUParticles2D::set_trail_length(float p_seconds) {
```

Comment:
Maybe `set_trail_duration` would be clearer than `set_trail_length`, which can sound like its related to distance and not time?

And then the property can be `trail_duration` instead of `trail_duration_secs` (durations are almost always seconds, though the doc will clarify it).

Reply:
Its related to both actually, its a length in time like in audio (song length). For the user its also a better naming because you know that changing this variable also changes the length.